.gitlab-ci.yml

```diff
@@ -867,3 +868,17 @@ audit_packages_scheduled:
       fi
   rules:
     - if: $CI_PIPELINE_SOURCE == "schedule"
+
+# EKS deployment
+deploy_eks:
+  trigger:
+    project: lg-public/identity-eks-control
+    # TODO change branch to main before merge
+    branch: lrappaport/deploy-eks
+  stage: deploy_eks
```

Comment:
Is there a specific job this stage depends on? Can we use [needs](https://docs.gitlab.com/ee/ci/yaml/#needs) to be more precise?

Reply:
Without the use of needs, the dependency is "all jobs in previous stages"

Reply:
If you look at https://gitlab.login.gov/lg/identity-idp/-/pipelines/120712 you can see it's currently the rightmost stage

Reply:
Right, sorry, I understand that aspect. This job is dependent on `deploy_production`?

I didn't quite understand the interaction with the deploy to "production" there with the description of deploying to different environments in the PR description.

Reply:
Yeah, we should seriously consider renaming the deploy_production job to deploy_reviewapp or something.  It's not production, and definitely can be confusing.

Reply:
Does `deploy_eks` depend on `deploy_production`? It seems like they could run concurrently?

Reply:
so i think the current `deploy_production` step is actually the "deploy to the `main` reviewapp" and is even more misnamed, if you look at this part

```
  environment:
    name: production
    deployment_tier: production
    url: https://$CI_ENVIRONMENT_SLUG.reviewapps.identitysandbox.gov
  rules:
    - if: $CI_COMMIT_BRANCH == "main" && $CI_PIPELINE_SOURCE == "push"
```

Reply:
and if that's the case, when this runs on main, we can deploy to eksdev and the main reviewapp in parallel

Reply:
which means we need build-idp-image in the review step, while trigger_devops runs in after_test, just dumping thoughts for now anyway

Reply:
i guess, my main requirements here are "everything" and if later on we add another step to the test phase i want that to automatically get picked up as a requirement? hence the stages thing?

Reply:
actually, doesn't the `main` reviewapp go away now? it would be the same as a continuously deployed dev env that this pipeline also adds?

https://idp.eksdev.identitysandbox.gov/

Reply:
if there's no objection i'm just gonna delete that step entirely

Reply:
I'd defer to @timothy-spencer on that aspect.

Reply:
I'm fine with nuking it.  Just want to make sure nobody else is using it.